internal/controller/nonadminbackup_controller_test.go

```diff
@@ -18,66 +18,636 @@ package controller
+					},
+					{
+						Type:    "Queued",
+						Status:  metav1.ConditionTrue,
+						Reason:  "BackupScheduled",
+						Message: "Created Velero Backup object",
+					},
+				},
+			},
+			result: reconcile.Result{Requeue: true},
+		}),
+		ginkgo.Entry("When called by Requeue(update NonAdminBackup VeleroBackupStatus), should update NonAdminBackup spec BackupSpec and Requeue", nonAdminBackupSingleReconcileScenario{
+			createVeleroBackup: true,
+			spec: nacv1alpha1.NonAdminBackupSpec{
+				BackupSpec: &v1.BackupSpec{},
+			},
+			priorStatus: &nacv1alpha1.NonAdminBackupStatus{
+				Phase:                 nacv1alpha1.NonAdminBackupPhaseCreated,
+				VeleroBackupName:      "nab-test-nonadminbackup-reconcile-7-c9dd6af01e2e2a",
+				VeleroBackupNamespace: "test-nonadminbackup-reconcile-7-oadp",
+				VeleroBackupStatus:    &v1.BackupStatus{},
+				Conditions: []metav1.Condition{
+					{
+						Type:               "Accepted",
+						Status:             metav1.ConditionTrue,
+						Reason:             "BackupAccepted",
+						Message:            "backup accepted",
+						LastTransitionTime: metav1.NewTime(time.Now()),
+					},
+					{
+						Type:               "Queued",
+						Status:             metav1.ConditionTrue,
+						Reason:             "BackupScheduled",
+						Message:            "Created Velero Backup object",
+						LastTransitionTime: metav1.NewTime(time.Now()),
+					},
+				},
+			},
+			status: nacv1alpha1.NonAdminBackupStatus{
+				Phase:                 nacv1alpha1.NonAdminBackupPhaseCreated,
+				VeleroBackupName:      "nab-test-nonadminbackup-reconcile-7-c9dd6af01e2e2a",
+				VeleroBackupNamespace: "test-nonadminbackup-reconcile-7-oadp",
+				VeleroBackupStatus:    &v1.BackupStatus{},
+				Conditions: []metav1.Condition{
+					{
+						Type:    "Accepted",
+						Status:  metav1.ConditionTrue,
+						Reason:  "BackupAccepted",
+						Message: "backup accepted",
+					},
+					{
+						Type:    "Queued",
+						Status:  metav1.ConditionTrue,
+						Reason:  "BackupScheduled",
+						Message: "Created Velero Backup object",
+					},
+				},
+			},
+			// TODO should not exit?
+			result: reconcile.Result{Requeue: true},
+		}),
+		ginkgo.Entry("When called by Requeue(update NonAdminBackup phase to new - invalid spec), should update NonAdminBackup phase to BackingOff and Requeue", nonAdminBackupSingleReconcileScenario{
+			spec: nacv1alpha1.NonAdminBackupSpec{
+				BackupSpec: &v1.BackupSpec{
+					IncludedNamespaces: []string{"not-valid"},
+				},
+			},
+			priorStatus: &nacv1alpha1.NonAdminBackupStatus{
+				Phase: nacv1alpha1.NonAdminBackupPhaseNew,
+			},
+			status: nacv1alpha1.NonAdminBackupStatus{
+				Phase: nacv1alpha1.NonAdminBackupPhaseBackingOff,
+			},
+			result: reconcile.Result{Requeue: true},
+		}),
+		ginkgo.Entry("When called by Requeue(update NonAdminBackup phase to BackingOff), should update NonAdminBackup Condition to Accepted False and stop with terminal error", nonAdminBackupSingleReconcileScenario{
```

Comment:
This test also fails if executed independently, I think this test should have passed. The counter value is the culprit here again I think. This test should pass independently as we are only testing one single reconcile loop functionality, right ? 
Lets also update the test description in line with the other suggestions.